src/renderer/decoder/polygonDecoder.js

```diff
@@ -27,27 +29,30 @@ export function decodePolygon (geometry) {
             const triangles = earcut(polygon.flat, polygon.holes);
             for (let k = 0; k < triangles.length; k++) {
```

Comment:
Could we check for overflow here and in `addLine`, and if necessary, make the buffer 2 times larger?

Reply:
This is an attempt to address this feature: https://github.com/CartoDB/carto-vl/pull/804